pgmpy/tests/test_models/test_MarkovChain.py

```diff
@@ -107,6 +107,31 @@ def test_add_variable_new(self):
         self.assertEqual(model.cardinalities['p'], 3)
         self.assertDictEqual(model.transition_models['p'], {})
 
+    def test_copy(self):
+        model = MC(['a', 'b'], [2, 2],[State('a', 0), State('b', 1)])
+        model.add_transition_model('a',{0: {0: 0.1, 1: 0.9}, 1: {0: 0.2, 1: 0.8}})
+        model.add_transition_model('b',{0: {0: 0.3, 1: 0.7}, 1: {0: 0.4, 1: 0.6}})
+        copy = model.copy()
+        self.assertIsInstance(copy, MC)
+        self.assertIsNot(copy, model)
+        self.assertEqual(model.variables, copy.variables)
+        self.assertEqual(model.cardinalities, copy.cardinalities)
+        self.assertEqual(model.transition_models, copy.transition_models)
+        self.assertEqual(model.state, copy.state)
+        model.add_variable('p', 1)
```

Comment:
@anchitjain1234 Try to group things together. Same as I have mentioned here: https://github.com/pgmpy/pgmpy/pull/591#discussion_r50214816